keps/262-ConfigurableFailurePolicy/README.md

```diff
@@ -276,6 +276,179 @@ spec:
+  replicatedJobs:
+  - name: leader
+    replicas: 1
+    template:
+      spec:
+        backoffLimit: 0
+        completions: 1
+        parallelism: 1
+        template:
+          spec:
+            restartPolicy: Never
+            containers:
+            - name: main
+              ...
+        podFailurePolicy:
+          rules:
+          - action: FailJob
+            name: DisruptionTarget # New field from the upstream change
+            onPodConditions:
+            - type: DisruptionTarget
+          - action: FailJob
+            name: ExitCode42 # New field from the upstream change
+            onExitCodes:
+              containerName: main
+              operator: In
+              values: [42]
+```
+
+While welcome, the upstream change will take time to be delivered. In the best 
+case scenario, the change will be included in Kubernetes 1.35 as alpha (ETA 
+January 2026) and become stable in 1.37 (ETA September 2026). A short-term 
+solution is to add the new field `spec.failurePolicy.rules[].onJobFailureMessagePatterns`, 
+which will allow the value of the `message` field in the Job failure condition 
+to be matched against a pattern. Although not designed to be machine-readable, 
+the message field is stable enough (no changes since its addition 2 years ago) 
+to serve as a short-term solution. Once the upstream solution is fully 
+delivered, the documentation and validation webhook of JobSet can be updated 
+to encourage new JobSet YAML files to use the long-term solution.
+
+**Short-Term solution: Example Failure Policy Configuration for this use case**:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: jobset-example
+spec:
+  failurePolicy:
+    maxRestarts: 5
+    rules:
+    - action: RestartJobSetAndIgnoreMaxRestarts
+      onJobFailureReasons:
+      - PodFailurePolicy
+      onJobFailureMessagePatterns: # New field
+      - ".*DisruptionTarget.*" # Failure message: Pod <pod namespace>/<pod name> has condition DisruptionTarget matching FailJob rule at index <index>
+    - action: FailJobSet
+      onJobFailureReasons:
+      - PodFailurePolicy
+      onJobFailureMessagePatterns: # New field
+      - ".*exit code 42.*" # Failure message: Container <container name> for pod <pod namespace>/<pod name> failed with exit code 42 matching FailJob rule at index <index>
+  replicatedJobs:
+  - name: leader
+    replicas: 1
+    template:
+      spec:
+        backoffLimit: 0
+        completions: 1
+        parallelism: 1
+        template:
+          spec:
+            restartPolicy: Never
+            containers:
+            - name: main
+              ...
+        podFailurePolicy:
+          rules:
+          - action: FailJob
+            onPodConditions:
+            - type: DisruptionTarget
+          - action: FailJob
+            onExitCodes:
+              containerName: main
+              operator: In
+              values: [42]
+```
+
```

Comment:
Not quite a big fan of doing regex on free form messages, but if there is a long term solution, then this is fine. please update the api as proposed in the Design Details section